pkg/controller/nstemplateset/nstemplatetier.go

```diff
@@ -15,23 +15,31 @@ import (
 	"sigs.k8s.io/kubefed/pkg/controller/util"
 )
 
+var cachedTierTemplates = map[string]*tierTemplate{}
```

Comment:
Let’s add a comment saying that the key here is a templateref name.
Or simply rename it to cachedTierTemplatesByTemplateRef (it’s probably too long name though)

Reply:
done 0e719f3